src/Microsoft.ML.FastTree/FastTree.cs

```diff
@@ -1887,7 +1880,7 @@ private void MakeBoundariesAndCheckLabels(out long missingInstances, out long to
                         missingInstances = cursor.BadFeaturesRowCount;
                     }
 
-                    ch.Check(totalInstances > 0, TrainerBase.NoTrainingInstancesMessage);
+                    ch.Check(totalInstances > 0, TrainerBase<IPredictor>.NoTrainingInstancesMessage);
```

Comment:
This is an unfortunate usage of generics...

Some thoughts:

* Should this message really be public?
* Should this message be moved to a non-generic class, so it can be accessed without the generic type?

Reply:
I noticed this too. Not quite sure... this field has about three total usages, and the idea of having the message standardize "hey you have no instances" I feel might be of limited enough worth that I might want to just nuke it altogether. I think it causes more confusion than it saves.